contracts/market/MarketsRegistry.sol

```diff
@@ -331,24 +342,59 @@ contract MarketsRegistry is OwnableUpgradeSafe, Proxiable, IMarketsRegistry {
         emit MarketDestroyed(address(market));
     }
 
+    function addFeeReceiver(
+        address receiver,
+        address _secondaryAddress,
+        uint8 _vaultPercentage
+    ) public onlyOwner {
+        require(receiver != address(0x0), "Invalid fee receiver address");
+        require(_secondaryAddress != address(0x0), "Invalid secondary address");
+
+        feeReceivers[receiver] = Receiver({
+            secondaryAddress: _secondaryAddress,
+            vaultPercentage: _vaultPercentage,
+            authorized: true
+        });
+    }
+
     /**
      * Allow owner to move tokens from the registry
      */
     function recoverTokens(IERC20 token, address destination)
         public
         override
-        onlyOwner
+        // onlyOwner
     {
         require(destination != address(0x0), "Invalid destination");
-
+        require(
+            feeReceivers[msg.sender].authorized 
+            || owner() == msg.sender,
+            "Sender address must be an authorized receiver or an owner"
+        );
         // Get the balance
         uint256 balance = token.balanceOf(address(this));
 
-        // Sweep out
-        token.safeTransfer(destination, balance);
+        if (msg.sender == owner()) {
```

Comment:
do we need a case when an owner can recover the tokens for themselves?

Reply:
I think yes